qfieldsync/core/cloud_api.py

```diff
@@ -734,6 +735,54 @@ def _clear_cloud_cookies(self, url: QUrl) -> None:
             self._nam.cookieJar().deleteCookie(cookie)
 
 
+    def ensure_localized_dataset_project(self) -> Optional[str]:
+        """
+        Ensures that the 'localized_datasets' project exists under the user's organization.
+
+        Returns:
+            str | None: The project ID of the 'localized_datasets' project, or None if creation failed.
+        """
+        try:
+         
+          
+            org_username = self.user_details.get("username")
+
+            if not org_username:
+                QgsMessageLog.logMessage("User not authenticated", "QFieldSync", Qgis.Warning)
+                return None
+            
+            existing_projects = self.get_projects_not_async()
+
+            for project in existing_projects:
+                if project.get("name") == "localized_datasets":
```

Comment:
you also need to project.get("owner") == org_username here, otherwise you'll risk picking a different owner's localized datasets.